src/base/settingsstorage.cpp

```diff
@@ -188,10 +188,13 @@ bool SettingsStorage::writeNativeSettings() const
         break;
     }
 
+    const Path newPath {nativeSettings->fileName()};
     if (nativeSettings->status() != QSettings::NoError)
+    {
+        Utils::Fs::removeFile(newPath);
```

Comment:
Are you sure the path isn't empty in case of error?
I've just confused by the previous code...

Reply:
>Are you sure the path isn't empty in case of error?

AFAIK the `newPath` would never be empty, it will always point to some path.